<a href="https://colab.research.google.com/github/varun0852/for-study/blob/master/hypothesis_testing.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import math
import scipy.stats as st
from scipy import stats
from scipy.stats import ttest_ind
from scipy.stats import jarque_bera
from scipy.stats import jarque_bera

Conduct a hypothesis test at significance level a = 0.05 to verify the claim made by the e-commerce company
Let us first define the hypothesis. If m is population mean, that is; mean delivery time then **bold text**
H0 : m ≥ 30 HA : m < 30

HO = null hypothesis(statement of no change)
HA = alternate hypothesis (statement of change)


In [ ]:
st.t.interval(confidence=0.99, df=len(time)-1,
              loc=np.mean(time))

(np.float64(24.342086816482354), np.float64(29.757913183517648))

In [ ]:
data = [16.0, 16.0, 30.0, 37.0, 25.0, 22.0, 19.0, 35.0, 27.0, 32.0,
34.0, 28.0, 24.0, 35.0, 24.0, 21.0, 32.0, 29.0, 24.0, 35.0,
28.0, 29.0, 18.0, 31.0, 28.0, 33.0, 32.0, 24.0, 25.0, 22.0,
21.0, 27.0, 41.0, 23.0, 23.0, 16.0, 24.0, 38.0, 26.0, 28.0]

In [ ]:
time = pd.Series(data = [16.0, 16.0, 30.0, 37.0, 25.0, 22.0, 19.0, 35.0, 27.0, 32.0,
34.0, 28.0, 24.0, 35.0, 24.0, 21.0, 32.0, 29.0, 24.0, 35.0,
28.0, 29.0, 18.0, 31.0, 28.0, 33.0, 32.0, 24.0, 25.0, 22.0,
21.0, 27.0, 41.0, 23.0, 23.0, 16.0, 24.0, 38.0, 26.0, 28.0])
print(time.mean())

27.05


In [ ]:
time.max()

41.0

In [ ]:
time.min()

16.0

In [ ]:
time.std()

6.2468453577047125

In [ ]:
# Population mean(Need to be provided by client)
pop_mean = 30

# Population Standard Deviation(Provided by client)
pop_std = 12.5

# sample mean
sample_mean = np.mean(data)

# length of sample data
n = len(data)

#Z_Score/Test which measures the difference in two population means
#when variance is known
# find the z-score = Number of Standard Deviation between Pop & Sample
z_score = (sample_mean - pop_mean)/(pop_std/np.sqrt(n))

# find the p - value
p_value = stats.norm.cdf(z_score)
# cdf = cumulative distribution function
print("The Z_Score value  is {} and the P_Value is {}".format(round(z_score , 4) , round(p_value , 4)))

The Z_Score value  is -1.4926 and the P_Value is 0.0678


We can see that the p-value is greater than 0.05, thus the null hypothesis is retained. Since 6.78% is greater than the significance value 5%, there is not enough evidence to reject null hypothesis. The null hypothesis is retained and it can be concluded that average delivery time is greater than or equal to 30.

when p_value < less than 0.05 then the probability of null hypothesis is 5%  and probability of alternate is 95% we accept alternate hypothesis  and if p_value is greter than 0.05 we accept null hypothesis